### DISCUSSION TITLE
Turning off parsing markup features like headers, image urls, html tags

### DISCUSSION THREAD

**Discussion by Nukiloco** [unifiedjs/unified#158] [Q&A]
Hi I was wondering. How do you turn off certain markup features in unified? For example I want to turn off headers, image urls, and html tags (in the markdown input) from being parsed at all and instead get treated like its regular text. Thanks!

**Comment by ChristianMurphy** [comment]
Sounds like https://github.com/micromark/micromark/discussions/31 and/or https://github.com/remarkjs/remark/issues/552

**Comment by Nukiloco** [comment]
remark-disable-tokenizers plugin doesn't work with the newer versions of remark which is sort of exactly what I want to use
is there any other way to disable block or inline tokens?

The extensions disable null method from https://github.com/micromark/micromark/commit/21fecde98dcf3529b88bbf8c9f6295cdf781b588 also doesn't seem to affect or turn off any markdown thats being parsed at all

**Reply by wooorm** [reply]
The extension should work, if it doesn’t, we can help better if you explain what you tried.

Perhaps also explaining more about what you want to achieve would help us answer your question. What *do* you want. If you don’t want URLs of images, what about the rest of the image syntax?

**Reply by Nukiloco** [reply]
Basically what I'm trying to achieve is multiple things.
First to disable these features:
- headers
- mail urls
- markdown image urls [aka ``![img](url)``]
- markdown urls (aka ``[test](url)`` which would turn into a url with the text of ``test``)
- tables
- code blocks

And then im also trying to add these features:
- auto embed detection for urls
- examples of auto embed detection:
  - https://via.placeholder.com/180.png would replace the url with an image instead
  - someurl/video.png would replace the url with an video instead
- user mentions (kind of like github/discord/twitter)

**Reply by wooorm** [reply]
Why do you want to turn most of markdown off — what is the point in using markdown then?
What about your users? They’d expect certain things to work, but then they don’t?
*Why* do you want to turn all these things off?

**Reply by Nukiloco** [reply]
I'm using markdown as the base for a comment system though as a personal preference I'm also turning off some features as well that wouldn't fit with the type of website I'm trying to make. Basically I want people to have a little bit of customization on their comments, but not to a full extent, kind of like how Discord does their comment system which has some features enabled for markdown but not all them.

**Reply by wooorm** [reply]
I strongly recommend you don’t do this — instead of arbitrarily breaking much of a users expectations you can also support all of commonmark and perhaps a bit of GFM on top.
But it’s possible. You can turn constructs off with micromark ([§ Case: turn off constructs](https://github.com/micromark/micromark#case-turn-off-constructs)) and you can wrap such extensions up in a remark plugin as well.

**Reply by Nukiloco** [reply]
Thing is, the website that I'm making won't say it supports markdown, it will just say that it supports custom formatting in comments. Basically I'm using a markdown parser to make my comment system.
